Classes/NSObject+RZImport.m

```diff
@@ -332,15 +332,28 @@ - (void)rzi_importValuesFromDict:(NSDictionary *)dict withMappings:(NSDictionary
     BOOL canOverrideImports = [self respondsToSelector:@selector( rzi_shouldImportValue:forKey: )];
     
     NSSet *ignoredKeys = [[self class] rzi_cachedIgnoredKeys];
-    
-    [dict enumerateKeysAndObjectsUsingBlock:^(NSString *key, id value, BOOL *stop) {
-    
+    NSOrderedSet *orderedKeys = [[self class] rzi_cachedOrderedKeys];
+
+    NSArray *dictKeys = dict.allKeys;
+
+    if ( orderedKeys.count > 0 ) {
+        dictKeys = [dictKeys sortedArrayUsingComparator:^NSComparisonResult(id obj1, id obj2) {
+            NSUInteger idx1 = [orderedKeys indexOfObject:obj1];
+            NSUInteger idx2 = [orderedKeys indexOfObject:obj2];
+
```

Comment:
Running this in 64-bit CodeRunner:

```objc
NSMutableArray *array = [NSMutableArray arrayWithCapacity:NSIntegerMax];
for ( NSUInteger i = 0; i < NSIntegerMax; i++ ) {
	[array addObject:@(i)];
}
```

yields this:

```
2015-06-11 17:25:31.575 Untitled[19511:9140511] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '*** -[__NSPlaceholderArray initWithCapacity:]: capacity (9223372036854775807) is ridiculous'
*** First throw call stack:
(
	0   CoreFoundation                      0x00007fff9103f03c __exceptionPreprocess + 172
	1   libobjc.A.dylib                     0x00007fff939ee76e objc_exception_throw + 43
	2   CoreFoundation                      0x00007fff90ee9b06 -[__NSPlaceholderArray initWithCapacity:] + 230
	3   CoreFoundation                      0x00007fff90f5dc0b +[NSMutableArray arrayWithCapacity:] + 43
	4   Untitled                            0x00000001045e9e42 main + 66
	5   libdyld.dylib                       0x00007fff94c3b5c9 start + 1
)
libc++abi.dylib: terminating with uncaught exception of type NSException
Abort trap: 6
```

Since the good part scrolls off the screen, here it is again:
> `-[__NSPlaceholderArray initWithCapacity:]: capacity (9223372036854775807) is ridiculous'`